lib/std/os/linux/start_pie.zig

```diff
@@ -56,13 +68,34 @@ fn getDynamicSymbol() [*]elf.Dyn {
             : [ret] "=r" (-> [*]elf.Dyn),
         ),
         // A simple `adr` is not enough as it has a limited offset range
-        .aarch64 => asm volatile (
+        .aarch64, .aarch64_be => asm volatile (
             \\ .weak _DYNAMIC
             \\ .hidden _DYNAMIC
             \\ adrp %[ret], _DYNAMIC
             \\ add %[ret], %[ret], #:lo12:_DYNAMIC
             : [ret] "=r" (-> [*]elf.Dyn),
         ),
+        // The CSKY ABI requires the gb register to point to the GOT. Additionally, the first
+        // entry in the GOT is defined to hold the address of _DYNAMIC.
+        .csky => asm volatile (
+            \\ mov %[ret], gb
+            \\ ldw %[ret], %[ret]
+            : [ret] "=r" (-> [*]elf.Dyn),
+        ),
+        .hexagon => asm volatile (
```

Comment:
From the Hexagon musl fork: https://github.com/quic/musl/blob/hexagon/arch/hexagon/crt_arch.h